### PR TITLE
Adding CDK to represent infrastructure needs.

### PR DESCRIPTION
**Pull request by ajaxx**
## Notes

This is a CDK to produce the infrastructure for a stack for this project.
The produced stack is agnostic and is currently producing non-sensical names because the stack should be self-contained.  Don't be alarmed, this is intentional.  You should be able to deploy the stack twice with no side effects.

## Testing

`cdk synth` - with manual inspection of the resultant CFN

`cdk deploy` - verified that the stack was created including API Gateway, Lambdas, and DynamoDB

### CODE REVIEW

**Comment by TheENGR** [comment]
Should this maybe have it's own repo? 

Not sure the defaultFiles repo is the right spot
And it might need to be private?

File: AWS/CDK/cdk.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "app": "node bin/cdk.js",
+  "context": {
+    "@aws-cdk/aws-apigateway:usagePlanKeyOrderInsensitiveId": true,
+    "@aws-cdk/core:enableStackNameDuplicates": "true",
+    "aws-cdk:enableDiffNoFail": "true",
+    "@aws-cdk/core:stackRelativeExports": "true",
```
Comment (TheENGR) [inline AWS/CDK/cdk.json:7]: Why are three `"true"` and the rest are `true`?

File: AWS/CDK/lib/lambda-stack.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      removalPolicy: cdk.RemovalPolicy.DESTROY
+    });
+
+    this.tblInfractions = new dynamodb.Table(this, 'DynamoDBInfractions', {
+      partitionKey: {
+        name: 'id',
+        type: dynamodb.AttributeType.STRING
+      },
+      readCapacity: 5,
+      writeCapacity: 5,
```
Comment (TheENGR) [inline AWS/CDK/lib/lambda-stack.js:50]: 5?

File: AWS/CDK/cdk.json (hunk 1 above)
Comment (ajaxx) [reply AWS/CDK/cdk.json:7]: these are the defaults for "cdk init" - I didn't bother to tweak them.

File: AWS/CDK/lib/lambda-stack.js (hunk 2 above)
Comment (ajaxx) [reply AWS/CDK/lib/lambda-stack.js:50]: So if you use provisioned IOPS you have to set this.  5 is a reasonable # of IOPS that will cause your dynamodb to throttle if traffic becomes too heavy.  For real use cases you might use request billing.

**Comment by ajaxx** [comment]
I agree on the location comment.  I'm not sure where to put it.  There is nothing "private" here.  It simply tells AWS to build a stack with a set of infrastructure.

File: AWS/CDK/cdk.json (hunk 1 above)
Comment (TheENGR) [reply AWS/CDK/cdk.json:7]: Super odd

**Pull request merged**